### PR TITLE
feat(nexmo.js): add support for Nexmo/Vonage SMS

### PR DESCRIPTION
**Pull request by ranrib**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Supports Vonage/Nexmo SMS

- **What is the current behavior?** (You can also link to an open issue here)
implements https://github.com/notifirehq/notifire/issues/26

- **What is the new behavior (if this is a feature change)?**

- **Other information**:

### CODE REVIEW

File: providers/nexmo/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Notifire Nexmo Provider
+
+A Nexmo sns provider library for [@notifire/core](https://github.com/notifirehq/notifire)
```
Comment (ranrib) [inline providers/nexmo/README.md:3]: Actually `sns` seems like the boilerplate bug @scopsy

File: providers/nexmo/src/lib/nexmo.provider.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const vonageResponseId : string = await new Promise((resolve, reject) => {
+      this.vonageClient.message.sendSms(
+        this.config.from,
+        options.to,
+        options.content,
+        {},
+        (err, responseData) => {
+          if (err) reject(err);
```
Comment (scopsy) [inline providers/nexmo/src/lib/nexmo.provider.ts:35]:
```suggestion
          if (err) return reject(err);
```

Missing a return here :)

File: providers/nexmo/src/lib/nexmo.provider.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  private vonageClient: Vonage;
+
+  constructor(
+    private config: {
+      apiKey: string;
+      apiSecret: string;
+      from: string;
+    }
+  ) {
+    this.vonageClient = new Vonage({apiKey: config.apiKey, apiSecret: config.apiSecret});
```
Comment (scopsy) [inline providers/nexmo/src/lib/nexmo.provider.ts:22]: Could you please run prettier in this file? 💅 @matthiez is taking a look at why it's not working on the precommit hook

File: providers/nexmo/README.md (hunk 1 above)
Comment (scopsy) [reply providers/nexmo/README.md:3]: @ranrib on it, thanks!

**Review by scopsy — COMMENTED** [review]
Looks amazing! Know a few folk who will be really excited for this one 🎉

If you could just run prettier on the provider file, until we will fix the `precommit` hook to run it automatically

**Comment by ranrib** [comment]
All fixed @scopsy

**Review by scopsy — APPROVED** [review]

**Pull request merged**